### ISSUE TITLE
NVDA doesn't read correctly at end of line in web documents

### ISSUE THREAD

**Issue by DataTriny** [nvaccess/nvda#8925]
### Steps to reproduce:
1. Open a web page containing a multi-line text field (aka textarea),
2. Write words without any line break,
3. Go to the end of the first line,
4. Go down and up the lines.

### Actual behavior:
When the cursor is at the end of the line, it reads the next (or previous) one, depending on where you were before. If the last read line was shorter and so the cursor is standing in the middle of the current line, then NVDA reports correct text.

### Expected behavior:
If the cursor is at the end of a line, it should read it, and not the one above or below.

#### NVDA Installed/portable/running from source:
NVDA is installed.

#### NVDA version:
NVDA version 2018.3.2.

#### Windows version:
Windows 10 Education, version 1803.

#### Name and version of other software in use when reproducing the issue:
- Firefox 63.0.1 (x64),
- Microsoft Edge

#### Does the issue still occur after restarting your PC?
Yes.

#### Have you tried any other versions of NVDA?
No.

**Comment by Adriani90** [comment]
@DataTriny, could you please  add an example website to the description which helps to reproduce this issue?

**Comment by DataTriny** [comment]
For sure. Simply go to: [https://github.com/nvaccess/nvda/issues/new] and use the comment body field., but it seems to work on any page.
The key thing is to type a long string of text without any line break so that it takes several lines.
Using the provided link, you can directly paste the following text on the appropriate field to highlight the issue:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa bbbbbbbbbbbbbbbbbbbbbb bbbbbbbbbbbbbbbbbbbbbb ccccccccccccccccccccccccccccccccccccccccccccccccccccccccc

It will span across three lines, so if you go to the end of the first one, and press the down arrow key, you should experience the issue as well.

**Comment by Adriani90** [comment]
Hmm, yes I can reproduce it. Si it occurs only if one line is shorter than the ogher. It seems that NVDA skips line 2, when at the end of line 1 I press down arrow. NVDA goes directly to line 3. Pressing up arrow brings me back to line 1. @leonardder could this be somehow related to principles used in #8479?

**Comment by Adriani90** [comment]
cc: @feerrenrut maybe you have any suggestions. This issue affects every browser, in most edit fields.

**Comment by Qchristensen** [comment]
As per https://github.com/nvaccess/nvda/issues/7078#issuecomment-619277941 This issue and #7078  appear to be the same issue.  I will close this issue, please move any further discussion to #7078.
